lib/src/domain/common/openalias.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:basic_utils/basic_utils.dart';
+
+String formatDomainName(String name) {
+  String formattedName = name;
+
+  if (name.contains(".")) {
+    formattedName = name.replaceAll("@", ".");
+  }
+
+  return formattedName;
+}
+
+Future<Map<String, String>> fetchXmrAddressAndRecipientName(String name) async {
```

Comment:
Better to create separated `class` for this info (like `OpenaliasRecord` or smth like that) with properties `name` and `address`. And rename the function for more short one (but keep it informative).